### PR TITLE
added travis api script

### PR DESCRIPTION
**Pull request by shahsaurabh0605**
Added the script to trigger `.travis.yml` file in `ruby-bench-docker`.

### CODE REVIEW

**Comment by tgxworld** [comment]
Can't we do this in Ruby instead?

**Comment by shahsaurabh0605** [comment]
Best way to do this should be in a script as indicated [here](https://docs.travis-ci.com/user/triggering-builds). For turning this to ruby we might need to use the `curb` which is the ruby wrapper for curl.

**Comment by tgxworld** [comment]
That is just one way of doing it. Essentially, what you're doing is just a POST request which can be done in many ways. One such way is to use the built in library in Ruby http://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTP.html

**Comment by shahsaurabh0605** [comment]
Made the changes in ruby. Have a look.

File: app/services/travis_script.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+require 'json'
+
+uri = URI.parse("https://api.travis-ci.org/repo/ruby-bench%2Fruby-bench-docker/requests")
+https = Net::HTTP.new(uri.host,uri.port)
+https.use_ssl = true
+request = Net::HTTP::Post.new(uri.path)
+request["Content-Type"] = 'application/json'
+request["Accept"] = 'application/json'
+request["Travis-API-Version"] = '3'
+request["Authorization"] = 'token smEOuqMAAUcwLHBnFjnJkA'
```
Comment (tgxworld) [inline app/services/travis_script.rb]: Does this token have to be a secret?

File: app/services/travis_script.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+request.body = {'request' => {'branch' => 'master'}}.to_json
+response = https.request(request)
```
Comment (tgxworld) [inline app/services/travis_script.rb]: What happens when the request returns a status other than `200`

File: app/services/travis_script.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+uri = URI.parse("https://api.travis-ci.org/repo/ruby-bench%2Fruby-bench-docker/requests")
```
Comment (tgxworld) [inline app/services/travis_script.rb]: What is with the encoding `%2` in the URL?

File: app/services/travis_script.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+https = Net::HTTP.new(uri.host,uri.port)
```
Comment (prathamesh-sonpatki) [inline app/services/travis_script.rb]: Nitpick but add a space between `uri.host` and `uri.port`.

Comment (prathamesh-sonpatki) [inline app/services/travis_script.rb]: Another nitpick, `https` doesn't sound right name. May be `http_client`?

**Comment by prathamesh-sonpatki** [comment]
@shahsaurabh0605 Please also check http://chris.beams.io/posts/git-commit/. Writing good commit messages is very important so in future, people will know why you added this script :smile:

File: app/services/travis_script.rb (hunk 3 above)
Comment (shahsaurabh0605) [inline app/services/travis_script.rb]: It's just the separator `%2F`. What of it?

File: app/services/travis_script.rb (hunk 2 above)
Comment (shahsaurabh0605) [inline app/services/travis_script.rb]: I did not get this. Currently we get a success. Do I need to check for something?

File: app/services/travis_script.rb (hunk 3 above)
Comment (tgxworld) [inline app/services/travis_script.rb]: Why do we need it `%2F` instead of just using`/` in string

File: app/services/travis_script.rb (hunk 2 above)
Comment (tgxworld) [inline app/services/travis_script.rb]: We can not always assume success, if the URI is wrong or if the api is down. You'll get a different status code.

Comment (shahsaurabh0605) [inline app/services/travis_script.rb]: What should I do for the cases when there is no success?

File: app/services/travis_script.rb (hunk 3 above)
Comment (shahsaurabh0605) [inline app/services/travis_script.rb]: It's just the same script I used from [here](https://docs.travis-ci.com/user/triggering-builds). So kept it that way.

**Pull request closed without merging**